working/macros/feature-specification.md

```diff
@@ -246,6 +246,41 @@ introspection APIs.
 These macros can fully introspect on any type reachable from the declarations
 they annotate, including introspecting on members of classes, etc.
 
+## Macro Instantiation
+
+Macros should be able to be able to be instantiated using only the libraries
+that are available to the macro definition library. This allows us to compile an
+app (or spawn an isolate) that can run macro applications for that macro using
+only its own dependencies.
+
+### Macro Constructor Parameters
+
+In order to facilitate this, parameters to Macro constructors are limited to
+a specific set of allowable types:
+
+- [Code][] or its subtypes
+- String
+- int, double, num
+- bool
+- null (really this just allows nullable types)
+- Uri
+- DateTime
```

Comment:
Why not just allow any value which can be safely deserialized by the isolate which is going to run the macro?
(And make it a compile-time error to specify any argument which can't)